### PR TITLE
Fixes #2798 by updating the table.

### LINKED ISSUE #2798: Incomplete class table for extends in ModelicaReference
The [class table](https://github.com/modelica/ModelicaStandardLibrary/blob/master/ModelicaReference/package.mo#L5543-L5612) on allowed inheritance misses the classes `operator function`, `operator record` and `expandable connector` of [MLS 3.2r2 page 68](https://modelica.org/documents/ModelicaSpec32Revision2.pdf#page=68).

### PR DESCRIPTION
**Pull request by dietmarw**
close #2798

### CODE REVIEW

File: ModelicaReference/package.mo [hunk 1]
```diff
@@ -5529,89 +5529,181 @@ equivalent to equations in the instantiated parent class are discarded.
...
 </table>
 
-<p>The specialized classes <code>package</code>,  <code>function</code>,  <code>type</code>, and  <code>record</code> can only be derived from their own kind <em>(e.g., a package can only be base class for packages. All other kinds of classes can use the import statement to use the contents of a package)</em>.</p>
+<p>The specialized classes <code>package</code>, <code>operator</code>,
+<code>function</code>, <code>type</code>, <code>record</code>,
+<code>operator record</code> and <code>expandable connector</code>
+can only be derived from their own kind
+[<em>(e.g., a package can only be base class for packages.
+All other kinds of classes can use the import statement to use the contents
+of a package)</em>] and from <code>class</code>.</p>
```
Comment (tobolar) [inline ModelicaReference/package.mo]: Why double parentheses? Why is this text emphasized?

Comment (dietmarw) [reply ModelicaReference/package.mo]: That's taken from the specs. So I kept it for now to be in sync.

**Comment by HansOlsson** [comment]
Note that italics in the spec isn't "emphasis" in the sense of emphasizing - but used to indicate non-normative text. That is already used in ModelicaReference - but I'm not sure if it is used consistently.

**Comment by dietmarw** [comment]
Probably not, but for now I would leave it and for the next version, when also the specs might have some more structure we can improve it based on that.

File: ModelicaReference/package.mo (hunk 1 above)
Comment (dietmarw) [reply ModelicaReference/package.mo]: Probably not, but for now I would leave it and for the next version, when also the specs might have some more structure we can improve it based on that.

**Pull request merged**